packages/backend-plugin-api/src/services/definitions/DiscoveryService.ts

```diff
@@ -60,4 +62,8 @@ export interface DiscoveryService {
    * like `https://backstage.example.com/api/catalog`
    */
   getExternalBaseUrl(pluginId: string): Promise<string>;
+
+  listFeatures?(): Promise<{
```

Comment:
It's alright to make this one required, see https://backstage.io/docs/overview/versioning-policy#changes-that-are-not-considered-breaking

I'd also be interested in making it clear through naming that this should include all features somehow, and maybe opening up for other info too in the future. Something like `getDeploymentInfo` or `getClusterInfo`?

I do also wonder if this is info that needs to be possible to subscribe to on some way. If doing feature detection then you might end up with some races at startup otherwise

Reply:
Should that sit on the discovery service though? What other info are you thinking might be added in the future? Feels like something that should sit one level up, like a `DeploymentMetadataService`? This function could then be added there, and the discovery api could be kept the same. Other ideas for methods on that new service might be, `listInstances`, and maybe cluster wide maintenance stuff like `updateTlsCertificates`, `restartInstance`.

I agree, subscribing would be really nice and could leverage signals for broadcasting new subscribers. It might also make sense to provide better HTTP client wrappers to handle any startup race conditions, even just simple retries could help.